src/pages/ProjectSettings.tsx

```diff
@@ -34,13 +37,30 @@ export const ProjectSettings: React.FC = () => {
     tab: ProjectSettingsTabRoutes;
   }>();
 
-  const { data } = useQuery<
+  // If the path includes an Object ID, this page represents a repo and we should not attempt to fetch a project.
+  const isRepo = validateObjectId(identifier);
+
+  const { data: projectData = null } = useQuery<
     ProjectSettingsQuery,
     ProjectSettingsQueryVariables
   >(GET_PROJECT_SETTINGS, {
+    skip: isRepo,
     variables: { identifier },
   });
 
+  const repoRefId =
+    projectData?.projectSettings?.projectRef?.repoRefId || identifier;
+  const useRepoSettings =
+    projectData?.projectSettings?.projectRef?.useRepoSettings;
```

Comment:
nit. destructure the common portions of these 2 imports

Reply:
I think since both of these variables require optional chaining, any attempt to use destructuring ends up being messier than this.